tests/v2/validation/rbac/secrets/rbac_secrets_test.go

```diff
@@ -0,0 +1,388 @@
+)
+
+type RbacSecretTestSuite struct {
+	suite.Suite
+	client            *rancher.Client
+	session           *session.Session
+	cluster           *management.Cluster
+	downstreamContext *wrangler.Context
+}
+
+func (rbs *RbacSecretTestSuite) TearDownSuite() {
+	rbs.session.Cleanup()
+}
+
+func (rbs *RbacSecretTestSuite) SetupSuite() {
+	rbs.session = session.NewSession()
+
+	client, err := rancher.NewClient("", rbs.session)
+	assert.NoError(rbs.T(), err)
+	rbs.client = client
+
+	log.Info("Getting cluster name from the config file and append cluster details in rb")
+	clusterName := client.RancherConfig.ClusterName
+	require.NotEmptyf(rbs.T(), clusterName, "Cluster name to install should be set")
+	clusterID, err := clusters.GetClusterIDByName(rbs.client, clusterName)
+	require.NoError(rbs.T(), err, "Error getting cluster ID")
+	rbs.cluster, err = rbs.client.Management.Cluster.ByID(clusterID)
+	assert.NoError(rbs.T(), err)
+}
+
+func (rbs *RbacSecretTestSuite) TestCreateSecretAsEnvVar() {
+	subSession := rbs.session.NewSession()
+	defer subSession.Cleanup()
+
+	tests := []struct {
+		role   rbac.Role
+		member string
+	}{
+		{rbac.ClusterOwner, rbac.StandardUser.String()},
+		{rbac.ClusterMember, rbac.StandardUser.String()},
+		{rbac.ProjectOwner, rbac.StandardUser.String()},
+		{rbac.ProjectMember, rbac.StandardUser.String()},
+		{rbac.ReadOnly, rbac.StandardUser.String()},
+	}
+
+	for _, tt := range tests {
+		rbs.Run("Validate secret creation for user with role "+tt.role.String(), func() {
+			log.Info("Create a project and a namespace in the project.")
+			adminProject, namespace, err := project.CreateProjectAndNamespace(rbs.client, rbs.cluster.ID)
+			assert.NoError(rbs.T(), err)
+
+			log.Infof("Create a standard user and add the user to a cluster/project role %s", tt.role)
+			newUser, standardUserClient, err := rbac.AddUserWithRoleToCluster(rbs.client, tt.member, tt.role.String(), rbs.cluster, adminProject)
+			assert.NoError(rbs.T(), err)
+			rbs.T().Logf("Created user: %v", newUser.Username)
+
+			log.Infof("As a %v, create a secret in the project %v", tt.role.String(), adminProject.Name)
+			secretData := map[string][]byte{
+				"hello": []byte("world"),
+			}
+			createdSecret, err := secret.CreateSecret(standardUserClient, rbs.cluster.ID, namespace.Name, secretData)
+			switch tt.role.String() {
+			case rbac.ClusterOwner.String(), rbac.ProjectOwner.String(), rbac.ProjectMember.String():
+				assert.NoError(rbs.T(), err, "failed to create secret")
+				log.Infof("As a %v, create a deployment using the secret as an environment variable.", tt.role.String())
+				createdDeployment, err := deployment.CreateDeployment(standardUserClient, rbs.cluster.ID, namespace.Name, 1, createdSecret.Name, "", true, false)
+				assert.NoError(rbs.T(), err, "failed to create deployment with secret")
+				err = charts.WatchAndWaitDeployments(standardUserClient, rbs.cluster.ID, namespace.Name, metav1.ListOptions{
```

Comment:
can we add WatchAndWaitDeployments in actions.workloads? we are importing charts just for this method and its a bit confusing that we are importing charts in this test..

Reply:
Good point, will add it to actions.workloads.deployment